File: CS/ChartLesson2/Controllers/ChartController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;

namespace ChartLesson2.Controllers {
    public class ChartController : Controller {
        public ActionResult Index() {
            using (var dbContext = new Models.NwindDbContext()) {
                var category =
                    (from c in dbContext.Categories
                     where c.CategoryName == "Beverages"
                     select c).First();
                var query = 
                    from p in dbContext.Products
                    where p.CategoryID == category.CategoryID
                    select p;
                return View(query.ToList());
            }
        }
    }
}

File: VB/ChartLesson2/Controllers/ChartController.vb
Imports System.Linq
Imports System.Web.Mvc

Namespace ChartLesson2.Controllers

    Public Class ChartController
        Inherits Controller

        Public Function Index() As ActionResult
            Using dbContext = New Models.NwindDbContext()
                Dim category =(From c In dbContext.Categories Where Equals(c.CategoryName, "Beverages") Select c).First()
                Dim query = From p In dbContext.Products Where p.CategoryID = category.CategoryID Select p
                Return View(query.ToList())
            End Using
        End Function
    End Class
End Namespace
